apps/task/tests/test_apis.py

```diff
@@ -0,0 +1,479 @@
+        self.assertEqual(content['data']['createTimeentry']['timeEntry']['task']['id'],
+                         str(self.input['task']))
+
+
+class UpdateTimeEntry(ChronoGraphQLTestCase):
+    def setUp(self):
+        self.mutation = '''mutation UpdateTimeEntry($input: TimeEntryUpdateInputType!){
+            updateTimeentry(timeEntry: $input){
+                errors {
+                    field
+                    messages
+                }
+                timeEntry {
+                    id
+                    description
+                    user{
+                        id
+                        email
+                    }
+                    task{
+                        id
+                    }
+                }
+                ok
+            }
+        }'''
+
+        self.input = {
+            "id": TimeEntryFactory.create().id,
+            "user": UserFactory.create().id,
+            "task": TaskFactory.create().id,
+            "date": "2020-11-10",
+            "startTime": "12:10:10"
+        }
+
+    def test_valid_taskentry_update(self):
+        response = self.query(
+            self.mutation,
+            input_data=self.input,
+        )
+
+        content = json.loads(response.content)
+        self.assertResponseNoErrors(response)
+        self.assertTrue(content['data']['updateTimeentry']['ok'], content)
+        self.assertIsNone(content['data']['updateTimeentry']['errors'], content)
+        self.assertEqual(content['data']['updateTimeentry']['timeEntry']['task']['id'],
+                         str(self.input['task']))
+        self.assertEqual(content['data']['updateTimeentry']['timeEntry']['id'],
+                         str(self.input['id']))
+
+
+class DeleteTimeEntry(ChronoGraphQLTestCase):
+    def setUp(self):
+        self.mutation = '''mutation DeleteTimeEntry($id: ID!){
+            deleteTimeentry(id: $id){
+                errors {
+                    field
+                    messages
+                }
+                timeEntry {
+                    id
+                    description
+                }
+                ok
+            }
+        }'''
+
+        self.timeEntry = TimeEntryFactory.create()
+        self.variables = {
+            "id": self.timeEntry.id,
+        }
+
+    def test_valid_timeentry_delete(self):
+        response = self.query(
+            self.mutation,
+            variables=self.variables,
+        )
+        content = json.loads(response.content)
+        self.assertTrue(content['data']['deleteTimeentry']['ok'], content)
+        self.assertIsNone(content['data']['deleteTimeentry']['errors'], content)
+        self.assertEqual(content['data']['deleteTimeentry']['timeEntry']['description'],
+                         self.timeEntry.description)
+        self.assertEqual(int(content['data']['deleteTimeentry']['timeEntry']['id']),
+                         self.timeEntry.id)
```

Comment:
Add invalid test cases as well. Under what constraints should tests fail? You need to handle those too. Rather than valid test cases, prioritize invalid test cases. The `clean or validate` methods should be tested. As a rule of thumb, all the custom functions should be tested.